timesketch/api/v1/resources/sigma.py

```diff
@@ -153,11 +213,374 @@ def post(self):
+            return abort(
+                HTTP_STATUS_CODE_BAD_REQUEST,
+                error_msg,
+            )
+        rule_uuid = parsed_rule.get("id")
+
+        title = parsed_rule.get("title")
+        description = parsed_rule.get("description")
+
+        # Query rules to see if it already exist and exit if found
+        sigma_rule_from_db = SigmaRule.query.filter_by(
+            rule_uuid=rule_uuid
+        ).first()
+        if sigma_rule_from_db:
+            if sigma_rule_from_db.rule_uuid == rule_uuid:
+                error_msg = (
+                    "Rule {0!s} was already found in the database".format(
+                        rule_uuid
+                    )
+                )
+                logger.debug(error_msg)
+                abort(HTTP_STATUS_CODE_BAD_REQUEST, error_msg)
+
+        sigma_rule = SigmaRule.get_or_create(
+            rule_uuid=rule_uuid,
+            rule_yaml=rule_yaml,
+            description=description,
+            title=title,
+            user=current_user,
+        )
+
+        if not sigma_rule:
+            abort(HTTP_STATUS_CODE_BAD_REQUEST, "No sigma rule object created")
+
+        sigma_rule.set_status(parsed_rule.get("status", "experimental"))
+        # query string is not stored in the database but we attach it to
+        # the JSON result here as it is added in the GET methods
+        sigma_rule.query_string = parsed_rule.get("es_query")
+
+        try:
+            db_session.add(sigma_rule)
+            db_session.commit()
+        except IntegrityError as e:
+            error_msg = "Problem adding Sigma rule {0!s}".format(e)
+            logger.error(error_msg)
+            abort(
+                HTTP_STATUS_CODE_CONFLICT,
+                error_msg,
+            )
+
+        return self.to_json(sigma_rule, status_code=HTTP_STATUS_CODE_CREATED)
+
+    @login_required
+    def put(self, rule_uuid):
+        """Handles update request to Sigma rules
+        Handels PUT API calls to /sigmarule/<string:rule_uuid>/ where the
+        rule_uuid is the primaray way to identify the rule.
+        The remaining attributes of the rule are provided in request itself.
+        If no `rule_yaml` is found in the reuqest, the method will fail as this
+        is required to parse the rule.
+
+        Args:
+            rule_uuid: uuid of the rule
+        Returns:
+            The updated sigma object in JSON (instance of
+            flask.wrappers.Response)
+        """
+
+        rule_yaml = request.json.get("rule_yaml")
+        if not rule_yaml:
+            abort(
+                HTTP_STATUS_CODE_BAD_REQUEST,
+                "Sigma parsing error: no yaml provided",
+            )
+        try:
+            parsed_rule = ts_sigma_lib.parse_sigma_rule_by_text(rule_yaml)
+        except ValueError as e:
+            error_msg = "Error parsing Sigma rule {0!s}: {1!s}".format(
+                rule_uuid, e
+            )
+            abort(HTTP_STATUS_CODE_BAD_REQUEST, error_msg)
```

Comment:
You're doing the same check in L 376, and returning a different error message: error_msg = "Sigma Rule Parsing error: {0!s}".format(e). Can we be consistent in the error messages that are returned in these situations across the changes in this PR?